language-server/modules/langserver-commons/src/main/java/org/ballerinalang/langserver/commons/SnippetContext.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.commons;
+
+/**
+ * Represents a context to track the number of placeholders.
+ *
+ * @since 2201.4.0
+ */
+public class SnippetContext {
+    private int placeholderCount = 1;
+
+    public SnippetContext() {
+    }
+
+    public int incrementAndGetPlaceholderCount() {
```

Comment:
`placeholderCount++` will return the current value of `placeholderCount` and then increcement it's value. But here the method name suggests otherwise. We should set initial value of `placeholderCount` to `0` and then do
```
return ++placeholderCount;
```
or
```
this.placeholderCount++;
return this.placeholderCount;
```